src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxNetworkInterface.cs

```diff
@@ -0,0 +1,158 @@
+        /// </summary>
+        /// <param name="interfaces">The Dictionary of existing interfaces.</param>
+        /// <param name="name">The name of the interface.</param>
+        /// <returns>The cached or new LinuxNetworkInterface with the given name.</returns>
+        private static LinuxNetworkInterface GetOrCreate(Dictionary<string, LinuxNetworkInterface> interfaces, string name)
+        {
+            LinuxNetworkInterface lni;
+            if (!interfaces.TryGetValue(name, out lni))
+            {
+                lni = new LinuxNetworkInterface(name);
+                interfaces.Add(name, lni);
+            }
+
+            return lni;
+        }
+
+        public override bool SupportsMulticast { get { return _supportsMulticast; } }
+
+        private static bool GetSupportsMulticast(string name)
+        {
+            string path = Path.Combine(NetworkFiles.SysClassNetFolder, name, NetworkFiles.FlagsFileName);
+            string fileContents = File.ReadAllText(path).Trim();
+            Interop.LinuxNetDeviceFlags flags = (Interop.LinuxNetDeviceFlags)Convert.ToInt32(fileContents, 16);
+            return (flags & Interop.LinuxNetDeviceFlags.IFF_MULTICAST) == Interop.LinuxNetDeviceFlags.IFF_MULTICAST;
+        }
+
+        public override IPInterfaceProperties GetIPProperties()
+        {
+            return _ipProperties;
+        }
+
+        public override IPInterfaceStatistics GetIPStatistics()
+        {
+            return new LinuxIPInterfaceStatistics(_name);
+        }
+
+        public override OperationalStatus OperationalStatus { get { return _operationalStatus; } }
+
+        public override string Id { get { throw new PlatformNotSupportedException(); } }
+
+        public override string Description { get { throw new PlatformNotSupportedException(); } }
+
+        public override long Speed
+        {
+            get
+            {
+                if (_speed.HasValue)
+                {
+                    return _speed.Value;
+                }
+                else
+                {
+                    throw new PlatformNotSupportedException();
+                }
+            }
+        }
+
+        public override bool IsReceiveOnly { get { throw new PlatformNotSupportedException(); } }
+
+        private static long? GetSpeed(string name)
+        {
+            try
+            {
+                string path = Path.Combine(NetworkFiles.SysClassNetFolder, name, NetworkFiles.SpeedFileName);
+                string contents = File.ReadAllText(path);
+                long val;
+                if (long.TryParse(contents, out val))
+                {
+                    return val;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+            catch (IOException) // Some interfaces may give an "Invalid argument" error when opening this file.
+            {
+                return null;
```

Comment:
Same as above: this is not consistent with the other parsers that fail if they can't obtain a row. We should decide on a consistent behavior.